dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/temp/TempFileResourceTest.java

```diff
@@ -0,0 +1,391 @@
+
+    final String fileName1 = "testFileName1" + UUIDGenerator.shorty() + ".png";
+    final String fileName2 = "testFileName2" + UUIDGenerator.shorty() + ".gif";
+    final Date date = new Date();
+    final FormDataContentDisposition fileMetaData = FormDataContentDisposition.name("testData").fileName(fileName1).creationDate(date)
+        .modificationDate(date).readDate(date).size(1222).build();
+
+    Response jsonResponse = resource.uploadTempResource(request, response, inputStream(), fileMetaData);
+
+    Map<String,List<DotTempFile>> dotTempFiles = (Map) jsonResponse.getEntity();
+    final DotTempFile dotTempFile1 = dotTempFiles.get("tempFiles").get(0);
+    
+    final RemoteUrlForm form = new RemoteUrlForm(
+        "https://raw.githubusercontent.com/dotCMS/core/master/dotCMS/src/main/webapp/html/images/skin/logo.gif", fileName2, null);
+
+    jsonResponse = resource.copyTempFromUrl(request, form);
+    dotTempFiles = (Map) jsonResponse.getEntity();
+    final DotTempFile dotTempFile2 = dotTempFiles.get("tempFiles").get(0);
+
+    final Map<String, Object> m = new HashMap<String, Object>();
+    m.put("stInode", contentType.id());
+    m.put("hostFolder", host.getIdentifier());
+    m.put("languageId", host.getLanguageId());
+    m.put("title", dotTempFile1.fileName);
+    m.put("fileAsset", dotTempFile1.id);
+    m.put("showOnMenu", "false");
+    m.put("sortOrder", 1);
+    m.put("description", "description");
+    m.put("testBinary", dotTempFile2.id);
+    Contentlet contentlet = new Contentlet(m);
+    contentlet = APILocator.getContentletAPI().checkin(contentlet, user, true);
+    
+    contentlet = APILocator.getContentletAPI().find(contentlet.getInode(), user, true);
+    
+    
+
+    assert (contentlet.getBinary("fileAsset").exists());
+    assert (contentlet.getBinary("fileAsset").getName().equals(fileName1));
+    assert (contentlet.getBinary("fileAsset").length() == 381411);
+    assert (contentlet.getBinary("testBinary").exists());
+    assert (contentlet.getBinary("testBinary").getName().equals(fileName2));
+    assert (contentlet.getBinary("testBinary").length() == 13695);
+
+  }
+
+  /**
+   * This test is for the TEMP_RESOURCE_ENABLED property
+   * If is set to false the temp resource is disabled and a 404 should be thrown.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void test_TempResourceAPI_TempResourceEnabledProperty() throws Exception{
+    final boolean tempResourceEnabledOriginalValue = Config.getBooleanProperty(TempFileAPI.TEMP_RESOURCE_ENABLED, true);
+    try {
+      final TempFileResource resource = new TempFileResource();
+
+      final HttpServletRequest request = mockRequest();
+
+      final String fileName = "test.png";
+      final HttpServletResponse response = new MockHttpResponse();
+
+      Config.setProperty(TempFileAPI.TEMP_RESOURCE_ENABLED, false);
+      final Date date = new Date();
+      final FormDataContentDisposition fileMetaData = FormDataContentDisposition.name("testData")
```

Comment:
I think part of this logic from line 366 to line 375 is repeated in many test and can be reused. In general I only see that a different property is set each time (in this case `Config.setProperty(TempFileAPI.TEMP_RESOURCE_ENABLED, false)`), which can be sent as a parameter